libs/payments-ui/src/l10n/l10n.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Children, useEffect, useState, ReactNode } from 'react';
+
+import { negotiateLanguages } from '@fluent/langneg';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
+import { ReactLocalization, LocalizationProvider } from '@fluent/react';
+
+const ftl: Record<string, URL> = {
+  'en-US': new URL('./en-US.ftl', import.meta.url),
+  fr: new URL('./fr.ftl', import.meta.url),
+};
+
+const DEFAULT_LOCALE = 'en-US';
+const AVAILABLE_LOCALES = {
+  'en-US': 'English',
+  fr: 'French',
+};
+
+async function fetchMessages(locale: string): Promise<[string, string]> {
+  const response = await fetch(String(ftl[locale]));
```

Comment:
cast to String feels a bit odd. Not sure if we want to use the `URL#href` prop here (or on L8-9 above) if we want a string version of the `URL` instance:
```diff
-  'en-US': new URL('./en-US.ftl', import.meta.url),
+  'en-US': new URL('./en-US.ftl', import.meta.url).href,
-  fr: new URL('./fr.ftl', import.meta.url),
+  fr: new URL('./fr.ftl', import.meta.url).href,
```